docs/dev/guidelines/client-tests.rst

```diff
@@ -54,7 +54,7 @@ The most basic test looks similar to this:
 
         it('should initialize', () => {
             someComponentFixture.detectChanges();
-            expect(SomeComponent).to.be.ok;
+            expect(SomeComponent).to.not.be.undefined;
```

Comment:
I agree that we should use stricter assertions wherever possible. Which is why I don't understand this change. `to.be.ok` should be a stricter assertion than `to.not.be.undefined` because the first assertion makes sure that the object is not null **and** not undefined. The second assertion allows null values.

Reply:
According to our code guidelines, `null` values do not exist in our system.
Technically it would be even better here to expect to deep equal to one specific component but that gets difficult and I would usually only recommend for simply types or objects.

Reply:
Yeah, I am still unsure about this though. Even if we shouldn't have null values in the system because of our code guidelines, I think it doesn't hurt that the test actually verifies this. So the test would basically assert that the coding guideline is actually followed. I agree that a deep equality would be the best assertion, but I also know that this is sometimes very cumbersome with complex objects.
This is more or less a minor thing though, so I can also ignore this and approve the PR if you would like

Reply:
I think this is always an issue if you just expect something due to guidelines. On the one hand, you do it to make it simpler. To not have to verify multiple things. On the other hand, there is always the "but what if someone does it anyway and it doesn't get detected?".
I think, though, that the latter thing is always an issue also with other parts of the code. There are always uncertainties because you can only know for sure that you did something wrong. You never know if you did something right for sure.

But maybe @krusche has an opinion on that?

Reply:
I think the example was already bad and that the change did not really make it better. How can a Component be ok? An assertion that is not undefined also does not really help for a good test case. The state of the component might be interesting, but then you would need to test some attributes of the component.

When looking through our client tests, I think we have way too many different options how to specify tests. So I would rather prefer to limit the guidelines to just one testing framework. Ideally we work with Jest only and try to avoid sinon, chai and jasmine when possible.

We had to make a lot of effort in the branch `chore/update-test-build` to migrate from Jasmin spies to Jest spies. While doing that we also had to get rid of some chai assertions and use the corresponding ones in "pure" Jest.

So my suggestion would be to update the testing guidelines to work with pure Jest mocks, spies and assertions!